src/setuptools_pyproject_migration/__init__.py

```diff
@@ -91,8 +91,50 @@ def _generate(self) -> Pyproject:
         if dependencies:
             pyproject["project"]["dependencies"] = dependencies
 
+        entrypoints = self._generate_entrypoints()
+
+        # GUI scripts and console scripts go separate in dedicated locations.
+        if "console_scripts" in entrypoints:
+            pyproject["project"]["scripts"] = entrypoints.pop("console_scripts")
+
+        if "gui_scripts" in entrypoints:
+            pyproject["project"]["gui-scripts"] = entrypoints.pop("gui_scripts")
+
+        # Anything left over gets put in entry-points
+        if entrypoints:
+            pyproject["project"]["entry-points"] = entrypoints
+
         return pyproject
 
+    def _parse_entrypoint(self, entrypoint: str) -> Dict[str, str]:
+        """
+        Extract the entry-point and name from the string.
+        """
+        # Format: 'hello-world = timmins:hello_world'
+        if "=" not in entrypoint:
+            raise ValueError("Entry point %r is not of the form 'name = target'" % entrypoint)
+
+        (name, target) = entrypoint.split("=", 1)
+        return {name.strip(): target.strip()}
+
+    def _generate_entrypoints(self) -> Dict[str, Dict[str, str]]:
+        """
+        Dump the entry-points for the project of the given type, if any.
+        """
+        entrypoints: Dict[str, Dict[str, str]] = {}
```

Comment:
This is somewhat a matter of personal preference, but I would employ the early return pattern here by first checking for the absence of the `entry_points` object:

```suggestion
        if not self.distribution.entry_points:
            return {}

        entrypoints: Dict[str, Dict[str, str]] = {}
```

and then you can skip the `if` statement below. I think it's easier to understand the code this way. (Not that less indentation correlates exactly to easy-to-understand in general, but I think it helps.)

Reply:
True, it avoids the march off the right-hand side of the screen.